fingerprint2.js

```diff
@@ -1388,14 +1388,9 @@
               return [p[0], p[1], mimeTypes].join('::')
             })})
         } else if (['canvas', 'webgl'].indexOf(component.key) !== -1) {
-          newComponents.push({key: component.key, value: component.value.join('~')})
+          if (component.value && typeof component.value.join === 'function') newComponents.push({key: component.key, value: component.value.join('~')})
```

Comment:
@Valve I think this should have an else case (just add `value: component.value`)

Reply:
sounds good, thanks